pkg/services/authn/identity.go

```diff
@@ -0,0 +1,179 @@
+	"github.com/grafana/grafana/pkg/services/user"
+)
+
+// NamespacedID builds a namespaced ID from a namespace and an ID.
+func NamespacedID(namespace string, id int64) string {
+	return fmt.Sprintf("%s:%d", namespace, id)
+}
+
+const (
+	NamespaceUser           = identity.NamespaceUser
+	NamespaceAPIKey         = identity.NamespaceAPIKey
+	NamespaceServiceAccount = identity.NamespaceServiceAccount
+	NamespaceAnonymous      = identity.NamespaceAnonymous
+	NamespaceRenderService  = identity.NamespaceRenderService
+)
+
+type Identity struct {
+	// OrgID is the active organization for the entity.
+	OrgID int64
+	// OrgName is the name of the active organization.
+	OrgName string
+	// OrgRoles is the list of organizations the entity is a member of and their roles.
+	OrgRoles map[int64]org.RoleType
+	// ID is the unique identifier for the entity in the Grafana database.
+	// It is in the format <namespace>:<id> where namespace is one of the
+	// Namespace* constants. For example, "user:1" or "api-key:1".
+	// If the entity is not found in the DB or this entity is non-persistent, this field will be empty.
+	ID string
+	// IsAnonymous
+	IsAnonymous bool
+	// Login is the shorthand identifier of the entity. Should be unique.
+	Login string
+	// Name is the display name of the entity. It is not guaranteed to be unique.
+	Name string
+	// Email is the email address of the entity. Should be unique.
+	Email string
+	// IsGrafanaAdmin is true if the entity is a Grafana admin.
+	IsGrafanaAdmin *bool
+	// AuthenticatedBy is the name of the authentication client that was used to authenticate the current Identity.
+	// For example, "password", "apikey", "auth_ldap" or "auth_azuread".
+	AuthenticatedBy string
+	// AuthId is the unique identifier for the entity in the external system.
+	// Empty if the identity is provided by Grafana.
+	AuthID string
+	// IsDisabled is true if the entity is disabled.
+	IsDisabled bool
+	// HelpFlags1 is the help flags for the entity.
+	HelpFlags1 user.HelpFlags1
+	// LastSeenAt is the time when the entity was last seen.
+	LastSeenAt time.Time
+	// Teams is the list of teams the entity is a member of.
+	Teams []int64
+	// idP Groups that the entity is a member of. This is only populated if the
+	// identity provider supports groups.
+	Groups []string
+	// OAuthToken is the OAuth token used to authenticate the entity.
+	OAuthToken *oauth2.Token
+	// SessionToken is the session token used to authenticate the entity.
+	SessionToken *usertoken.UserToken
+	// ClientParams are hints for the auth service on how to handle the identity.
+	// Set by the authenticating client.
+	ClientParams ClientParams
+	// Permissions is the list of permissions the entity has.
+	Permissions map[int64]map[string][]string
+	// IDToken is a signed token representing the identity that can be forwarded to plugins and external services.
+	// Will only be set when featuremgmt.FlagIdForwarding is enabled.
+	IDToken string
```

Comment:
Should we call this id token or do you have a better name @Jguer @gamab ?

Reply:
should be ok :shrug: 
we're still on time to change it needed